azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -41,15 +101,242 @@ def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_ev
         else:
             logger.error(msg)
         if send_event:
-            add_event(op=op, message=msg, is_success=is_success)
+            add_event(op=op, message=msg, is_success=is_success, log_event=False)
+
+    @staticmethod
+    def __get_policy_enforcement_enabled():
+        """
+        Policy will be enabled if (1) policy file exists at the expected location and (2) the conf flag "Debug.EnableExtensionPolicy" is true.
+        """
+        return conf.get_extension_policy_enabled() and os.path.exists(conf.get_policy_file_path())
+
+    @property
+    def policy_enforcement_enabled(self):
+        return self._policy_enforcement_enabled
 
     @staticmethod
-    def is_policy_enforcement_enabled():
+    def __get_policy():
+        """
+        Read customer-provided policy JSON file, load and return as a dict.
+        Policy file is expected to be at conf.get_policy_file_path(). Note that this method should only be called
+        after verifying that the file exists (currently done in __init__).
         """
-        Check whether user has opted into policy enforcement feature.
-        Caller function should check this before performing any operations.
+        with open(conf.get_policy_file_path(), 'r') as f:
+            _PolicyEngine._log_policy_event(
+                "Policy enforcement is enabled. Enforcing policy using policy file found at '{0}'.".format(
+                    conf.get_policy_file_path()))
```

Comment:
Is it worth logging/sending telemetry with the provided policy (either before or after parsing)?

We don't want to flood the log, but we only check policy when there's a new extensions goal state to process, so I think it may be useful to add this logging/telemetry to aid in debugging. Right now, if a customer raises an incident that their policy was incorrectly applied, we won't know what the state of the policy file was at the time of the issue.

@narrieta, @mgunnala  what do you think?

Reply:
I think it could be worth logging the policy immediately after reading it (before parsing), it would be helpful especially in cases where the customer unexpectedly changes the file during goal state processing.

Reply:
Added logging on read